packages/@lwc/features/src/runtime.ts

```diff
@@ -6,41 +6,57 @@
  */
 import features, { FeatureFlagLookup, FeatureFlagValue } from './flags';
 import { create, isFalse, isTrue, isUndefined } from '@lwc/shared';
+import { getGlobalThis } from './global-this';
 
-const runtimeFlags: FeatureFlagLookup = create(null);
+const _globalThis: any = getGlobalThis();
+if (!_globalThis.lwcRuntimeFlags) {
+    Object.defineProperty(_globalThis, 'lwcRuntimeFlags', { value: create(null) });
+}
+
+const runtimeFlags: FeatureFlagLookup = _globalThis.lwcRuntimeFlags;
 
 // This function is not supported for use within components and is meant for
 // configuring runtime feature flags during app initialization.
 function setFeatureFlag(name: string, value: FeatureFlagValue) {
     const isBoolean = isTrue(value) || isFalse(value);
     if (!isBoolean) {
-        const message = `Invalid ${typeof value} value specified for the "${name}" flag. Runtime feature flags can only be set to a boolean value.`;
+        const message = `Failed to set the value "${value}" for the runtime feature flag "${name}". Runtime feature flags can only be set to a boolean value.`;
         if (process.env.NODE_ENV === 'production') {
             // eslint-disable-next-line no-console
             console.error(message);
+            return;
         } else {
             throw new TypeError(message);
         }
     }
-    if (!isUndefined(features[name])) {
-        runtimeFlags[name] = value;
-    } else {
+    if (isUndefined(features[name])) {
         // eslint-disable-next-line no-console
         console.warn(
-            `LWC feature flag "${name}" is undefined. Possible reasons are that 1) it was misspelled or 2) it was removed from the @lwc/features package.`
+            `Failed to set the value "${value}" for the runtime feature flag "${name}" because it is undefined. Possible reasons are that 1) it was misspelled or 2) it was removed from the @lwc/features package.`
         );
+        return;
+    }
+    if (process.env.NODE_ENV === 'production') {
+        // Disallow the same flag to be set more than once in production
+        const runtimeValue = runtimeFlags[name];
+        if (!isUndefined(runtimeValue)) {
+            // eslint-disable-next-line no-console
+            console.error(
+                `Failed to set the value "${value}" for the runtime feature flag "${name}". "${name}" has already been set with the value "${runtimeValue}".`
+            );
+            return;
+        }
+        Object.defineProperty(runtimeFlags, name, { value });
```

Comment:
What is the value of using `Object.defineProperty` over a simple property assignment even if we protect the assignment in the previous line?

Reply:
I think what @ekashida is doing here is saying that in prod we do not allow a flag to change it state after it was set.

Reply:
We're just adding a helpful error message when the flag is being set through `setFeatureFlag`. We could move that into a setter but I'm less sympathetic to someone trying to set the flag directly.